Classes/NodeCreationHandler/CreationDialogPropertiesCreationHandler.php

```diff
@@ -27,37 +25,22 @@ class CreationDialogPropertiesCreationHandler implements NodeCreationHandlerInte
 {
     /**
      * @Flow\Inject
-     * @var PropertyMapper
+     * @var NodePropertyConversionService
      */
-    protected $propertyMapper;
+    protected $nodePropertyConversionService;
 
-    /**
-     * @param NodeInterface $node The newly created node
-     * @param array $data incoming data from the creationDialog
-     * @return void
-     * @throws PropertyException | SecurityException
-     */
     public function handle(NodeInterface $node, array $data): void
     {
-        $propertyMappingConfiguration = $this->propertyMapper->buildPropertyMappingConfiguration();
-        $propertyMappingConfiguration->forProperty('*')->allowAllProperties();
-        $propertyMappingConfiguration->setTypeConverterOption(PersistentObjectConverter::class, PersistentObjectConverter::CONFIGURATION_OVERRIDE_TARGET_TYPE_ALLOWED, true);
-
-        foreach ($node->getNodeType()->getConfiguration('properties') as $propertyName => $propertyConfiguration) {
+        foreach ($data as $propertyName => $propertyValue) {
+            $propertyConfiguration = $node->getNodeType()->getConfiguration('properties')[$propertyName];
             if (!isset($propertyConfiguration['ui']['showInCreationDialog']) || $propertyConfiguration['ui']['showInCreationDialog'] !== true) {
                 continue;
             }
             $propertyType = TypeHandling::normalizeType($propertyConfiguration['type'] ?? 'string');
-            if (!isset($data[$propertyName])) {
+            if ($propertyValue === null || ($propertyValue === '' && !TypeHandling::isSimpleType($propertyType))) {
```

Comment:
You introduced this `null` skipping here but i makes sense i guess.

I am more curios about the  `$propertyValue === '' && !TypeHandling::isSimpleType($propertyType)` part because it seems to have gone missing in 9.0 after the ESCR [adjustments](https://github.com/neos/neos-ui/commit/1498c8b4c6d84faf993dac1dd3756ad20993e4dd#diff-773a367aeb44c8b1dc0134079eaa8f3483e76936c98682d720234ce87a6f44a8L55-L57).

Originally it was introduced to fix https://github.com/bwaidelich/Wwwision.Neos.CreationDialogProperties/issues/10

But when using the `nodePropertyConversionService` we seem to get this `$rawValue !== ''` check for free in:

https://github.com/neos/neos-ui/blob/01ac51c52e66a2ad6fb9d9dcbff9623739e5da14/Classes/Domain/Service/NodePropertyConversionService.php#L87

So not skipping this property would write an empty string to it but that is fine and seems to be expected because of this check 😅 

https://github.com/neos/neos-development-collection/blob/eada1dbfb569de6e3932b26b4b3c630401020e05/Neos.ContentRepository/Classes/Domain/Model/Node.php#L924-L931

we might as well keep it for 8.3 like this bu my brain is exploding when i think about 9.0 :D

Reply:
Well, it is what the `isset()` checks does, but not "as strict", i.e. anything that is `null` or not a non-empty string representing some non-literal, is decteted.

Reply:
Okay and regarding 9.0 i created a followup https://github.com/neos/neos-ui/issues/3755